File: Python/juegoconherencia/Espadachin.py
from Personaje import Personaje
import random

class Espadachin(Personaje):
    
    def __init__(self, nombre: str, vida: int):
        super().__init__(nombre, vida)
    
    @classmethod    
    def Constructor(cls, nombre):
        nuevoLadron = cls(nombre, 100)
        return nuevoLadron
    
    def atacar(self, objetivo, dano: int):
        if random.randint(0,1) == 1:
            dano = self.__habilidad_especial__(dano)
        super().atacar(objetivo, dano)
    
    def __habilidad_especial__(self, dano):
        return dano*20

File: Java/juegoconherencia/src/Espadachin.java
import java.util.Random;

public class Espadachin extends Personaje{
    
    public Espadachin(String nombre, int vida) {
        super(nombre, vida);
    }

    public Espadachin(String nombre) {
        this(nombre, 100);
    }

    @Override
    public void atacar(Personaje objetivo, int dano) {
        if(new Random().nextBoolean()){
            dano  = this.hablidadEspecial(dano);
        }
        super.atacar(objetivo, dano);
    }

    @Override
    protected Integer hablidadEspecial(int dano) {
        return dano*20;
    }
}
